libraries/testing/src/with-pulse.js

```diff
@@ -0,0 +1,218 @@
+const assert = require('assert');
+const {QlobberTrue} = require('qlobber');
+const EventEmitter = require('events');
+const debug = require('debug');
+
+module.exports = ({helper, skipping, namespace}) => {
+  let client;
+  const debugPulseAssertion = debug('withPulse');
+
+  suiteSetup('withPulse', async function() {
+    if (skipping && skipping()) {
+      return;
+    }
+
+    await helper.load('cfg');
+    client = new FakeClient(namespace);
+    helper.load.inject('pulseClient', client);
+
+    const matchingMessageExists = (exchange, check) =>
+      client.messages.some(message =>
+        (!exchange || message.exchange.endsWith(exchange)) &&
+        (!check || check(message)));
+
+    helper.assertPulseMessage = (exchange, check) => {
```

Comment:
I notice that a lot of the checks we've replace used to have a `deepEqual`. I see that is not possible now with this more generic form but I wonder if we can have some way to at least log the payload of the message or something? Does not block landing this.

Reply:
Yeah, that annoyed me at one point and I fell back to sticking in some `console.logs`.  `_.isEqual` works, but it doesn't give any diff when it fails.  I'm not sure how to fix that..